test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/testutil/MockIdentityProviderUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.testutil;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+import software.amazon.awssdk.identity.spi.ResolveIdentityRequest;
+
+public class MockIdentityProviderUtil {
+
+    public static IdentityProvider<AwsCredentialsIdentity> mockIdentityProvider() {
+        IdentityProvider<AwsCredentialsIdentity> mockIdentityProvider = mock(IdentityProvider.class);
+        setup(mockIdentityProvider);
+        return mockIdentityProvider;
+    }
+
+    public static void setup(IdentityProvider<AwsCredentialsIdentity> mockIdentityProvider) {
+        when(mockIdentityProvider.resolveIdentity(any(ResolveIdentityRequest.class))).thenAnswer(invocation -> {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException ie) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(ie);
+            }
+            return CompletableFuture.completedFuture(AwsBasicCredentials.create("foo", "bar"));
```

Comment:
should these `foo` `bar` value be configurable for better reuse on other tests?

Reply:
We could but I don't see much value on complicating the code, those are random values that are just needed to avoid NPE and not actually used otherwise.